### PR TITLE
mgr/dashboard: Add frontend code documentation

### PR DESCRIPTION
**Pull request by epuertat**
Adds [compodoc](https://github.com/compodoc/compodoc) as code documentation auto-tool for Angular frontend code.

image

Fixes: http://tracker.ceph.com/issues/36243
Signed-off-by: Ernesto Puerta <epuertat@redhat.com>

- [x] References tracker ticket
- [x] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/pybind/mgr/dashboard/HACKING.rst [hunk 1]
```diff
@@ -65,6 +65,15 @@ Run ``npm run build`` to build the project. The build artifacts will be
 stored in the ``dist/`` directory. Use the ``--prod`` flag for a
 production build (``npm run build -- --prod``). Navigate to ``https://localhost:8443``.
 
+Build the Code Documentation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Run ``npm run doc-build`` to generate code docs in the ``documentation/``
+directory. To make them accesible locally for a web browser, run
+``npm run doc-serve`` and they will become available at ``http:://localhost:8444``.
```
Comment (s0nea) [inline src/pybind/mgr/dashboard/HACKING.rst]: Is the second ':' on purpose?

```suggestion
``npm run doc-serve`` and they will become available at ``http://localhost:8444``.
```

Comment (LenzGr) [reply src/pybind/mgr/dashboard/HACKING.rst]: @epuertat - can you please fix the URL accordingly?

Comment (epuertat) [reply src/pybind/mgr/dashboard/HACKING.rst]: Good catch, @s0nea! Thanks!

**Comment by epuertat** [comment]
Closed by accident (while cleaning up origin branches).

**Pull request merged**